libs/components/forms/src/lib/modules/input-box/input-box.component.ts

```diff
@@ -51,34 +51,34 @@ export class SkyInputBoxComponent implements OnInit {
    * @default false
    */
   @Input()
-  public disabled: boolean;
+  public disabled: boolean | undefined;
 
-  public hostInputTemplate: TemplateRef<unknown>;
+  public hostInputTemplate: TemplateRef<unknown> | undefined;
 
-  public hostButtonsTemplate: TemplateRef<unknown>;
+  public hostButtonsTemplate: TemplateRef<unknown> | undefined;
 
-  public hostButtonsInsetTemplate: TemplateRef<unknown>;
+  public hostButtonsInsetTemplate: TemplateRef<unknown> | undefined;
 
-  public hostButtonsLeftTemplate: TemplateRef<unknown>;
+  public hostButtonsLeftTemplate: TemplateRef<unknown> | undefined;
 
-  public formControlHasFocus: boolean;
+  public formControlHasFocus = false;
 
-  public hostIconsInsetTemplate: TemplateRef<unknown>;
+  public hostIconsInsetTemplate: TemplateRef<unknown> | undefined;
 
-  public hostIconsInsetLeftTemplate: TemplateRef<unknown>;
+  public hostIconsInsetLeftTemplate: TemplateRef<unknown> | undefined;
 
   @ContentChild(FormControlDirective)
-  public formControl: FormControlDirective;
+  public formControl: FormControlDirective | undefined;
 
   @ContentChild(FormControlName)
-  public formControlByName: FormControlName;
+  public formControlByName: FormControlName | undefined;
 
   @ContentChild(NgModel)
-  public ngModel: NgModel;
+  public ngModel: NgModel | undefined;
 
   public get hasErrorsComputed(): boolean {
```

Comment:
I'm calling this function out. This is used by the template - which is normally not a good thing. I tried to get rid of it; however, the calculation here requires us to know and react to the touched/dirty state of the form control. That stat is not something that Angular gives events for when it changes and because of that I can't move this logic elsewhere as we need to react accordingly when that changes.